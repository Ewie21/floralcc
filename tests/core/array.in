int h = [8; '6', '5', '4', '3', '2', '1', '0'];
int y = [1; 10];
y[0] = '6';
int j = '1';
put(y[0]);
put('\n');

put(*h);
put(*(h - 8));
put(h[2]);
put(h[3]);
put(h[4]);
put(h[5]);
put(h[6]);
put('\n');

*h = '7';
*(h - 8) = '6';
h[2] += 1;
h[3] += 1;
h[4] += 1;
h[5] += 1;
h[6] += 1;

put(h[0]);
put(h[1]);
put(h[2]);
put(h[3]);
put(h[4]);
put(h[5]);
put(h[6]);
put('\n');

h[0] = 'b';
put(h[0]);
put(h[1]);
put('\n');
put(j);
put(h[6]);

void print_array(int ptr) {
    int anchor = ptr;
    while (*ptr != 0) {
        put(*ptr);
        ptr -= 1;
    }
    ptr = anchor;
}

put('\n');
put('\n');
print_array(h);
