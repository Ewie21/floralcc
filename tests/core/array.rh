int h = ['6', '5', '4', '3'];
int j = '1';
put(*h);
put(*(h - 8));
put(*(h - 16));
put(*(h - 24));
put('2');
put(j);
