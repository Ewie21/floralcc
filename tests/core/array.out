6
6543210
7654321
b6
11

b654321
